src/indicators/r003.rs

```diff
@@ -0,0 +1,37 @@
+use chrono::DateTime;
+use serde_json::{Map, Value};
+
+use crate::indicators::{set_result, Calculate, Indicators, Settings};
+
+#[derive(Default)]
+pub struct R003 {
+    threshold: usize,
+}
+
+impl Calculate for R003 {
+    fn new(settings: &mut Settings) -> Self {
+        Self {
+            threshold: std::mem::take(&mut settings.R003)
+                .unwrap_or_default()
+                .threshold
+                .unwrap_or(15),
+        }
+    }
+
+    fn fold(&self, item: &mut Indicators, release: &Map<String, Value>, ocid: &str) {
+        if let Some(Value::Object(tender)) = release.get("tender")
+            && let Some(Value::String(procurement_method)) = tender.get("procurementMethod")
+            && let Some(Value::Object(tender_period)) = tender.get("tenderPeriod")
+            && let Some(Value::String(end_date)) = tender_period.get("endDate")
+            && let Some(Value::String(start_date)) = tender_period.get("startDate")
+        {
+            let start_date = DateTime::parse_from_rfc3339(start_date).unwrap();
```

Comment:
@jpmckinney I've read in our guide that we shouldn't `panic!` so suggestions are welcome about how to handle this without `unwrap()`

Reply:
Indeed, the command should not fail simply because of one bad value, for example.

If there is a sensible default value, we typically use unwrap_or_default, unwrap_or_else (if the default requires computation) or unwrap_or (if the default is a simple value) https://doc.rust-lang.org/nightly/core/result/enum.Result.html (chrono's ParseResult [is the same as](https://docs.rs/chrono/0.4.11/chrono/format/type.ParseResult.html) Rust's Result).

If there isn't a possible (or even creative) default, then we use `if let` statements like (continuing from the earlier if statement):

```rust
    && let Ok(start_date) = DateTime::parse_from_rfc3339(start_date)
    && let Ok(end_date) = DateTime::parse_from_rfc3339(end_date)
    && procurement_method = "open"
    && (end_date - start_date).num_days() < self.threshold.try_into().unwrap_or(15)
{
```